### PR TITLE
  Added the option to create password protected .zip files in New-WindowsFromGoldenImage for KVM type images

### PR DESCRIPTION
**Pull request by andreitira**

### CODE REVIEW

File: WinImageBuilder.psm1 [hunk 1]
```diff
@@ -616,6 +615,33 @@ function Compress-Image {
     Write-Output "MaaS image is ready and available at: $ImagePath"
 }
 
+function Create-ProtectedZip {
```
Comment (ader1990) [inline WinImageBuilder.psm1]: Leave the method name unchanged Start-WindowsCommand and use the alias param when you use it.

File: WinImageBuilder.psm1 [hunk 2]
```diff
@@ -1193,6 +1221,12 @@ function New-WindowsFromGoldenImage {
                 Write-Output "Converting VHD to QCow2"
                 Convert-VirtualDisk $WindowsImageVHDXPath $Qcow2ImagePath "qcow2"
                 Remove-Item -Force $WindowsImageVHDXPath
+                If ($zipPassword){
```
Comment (ader1990) [inline WinImageBuilder.psm1]: if*

File: WinImageBuilder.psm1 [hunk 3]
```diff
@@ -1193,6 +1221,12 @@ function New-WindowsFromGoldenImage {
                 Write-Output "Converting VHD to QCow2"
                 Convert-VirtualDisk $WindowsImageVHDXPath $Qcow2ImagePath "qcow2"
                 Remove-Item -Force $WindowsImageVHDXPath
+                If ($zipPassword){
+                    $final_path=$barePath + ".zip"
```
Comment (ader1990) [inline WinImageBuilder.psm1]: zipPath

Comment (ader1990) [reply WinImageBuilder.psm1]: plus make sure indentation is correct

**Review by ader1990 — CHANGES_REQUESTED** [review]
a few naming and PowerShell style issues, the logic is good

**Comment by costingalan** [comment]
@andreitira: I think we need to write the password after the zip was created so that it will be easy to copy it.

**Comment by costingalan** [comment]
I suggest you add:

- [ ] Information in the README.md file.
- [ ] Possibility to create password protected files when using New-WindowsOnlineImage, New-MAASImage and also New-WindowsCloudImage.
- [ ] Add the $zipPassword as an optional parameter to the above functions.

**Comment by andreitira** [comment]
Made all the requested changes :)
Please take a look at my implementation for New-WindowsCloudImage. Im a bit uncertain about that one.

**Pull request closed without merging**